Add documentation about creating tables using sqlalchemy schemas; Close #234

empicano opened the pull request:
Hi there,

First of all, thank you for your work on this great library.

I had to search for a while in the issues to find out how to create tables based on SQLAlchemy schemas, so I added a section about it in the docs, based on #234. I'm open to all comments 😊 

~ Felix

empicano commented:
[The documentation](https://www.encode.io/databases/database_queries/) has a section "Table declarations", followed by the section "Queries", which assumes that those tables now exist.

Personally, I got stuck there because I didn't know how to actually create the tables. I think a note here would be helpful.

I'm not dead set on anything, and I agree that alembic is a more solid choice. I'll add a note about this.

For experiments, and falling back to SQLAlchemy, is there a way to "extract" the engine from databases? I think it would be neat to avoid opening a new connection just to create the tables.

rafalp commented:
Will the approach you are proposing also handle the constraints and keys? Or will it just create tables?

rafalp commented on docs/database_queries.md:
```diff
@@ -24,9 +24,48 @@ notes = sqlalchemy.Table(
 
+## Creating tables
+
+Due to databases wrapping around multiple other packages, you can't use [the usual SQLAlchemy core way to create tables](https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.MetaData.create_all). Luckily, there's a neat workaround using schema compilation:
```
We should be more specific why some things are not possible and struggle for neutral language:

```suggestion
Databases don't use SQLAlchemy's engines for database access, making [the usual SQLAlchemy core way to create tables with `create_all`](https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.MetaData.create_all) not available. To work around this you can build `TABLE CREATE` queries manually:
```

rafalp commented on docs/database_queries.md:
````diff
@@ -24,9 +24,48 @@ notes = sqlalchemy.Table(
+await database.disconnect()
+```
+
+Note that this way of creating tables is only useful for experiments. In a production environment, you'd rather use something like [Alembic](https://alembic.sqlalchemy.org/en/latest/), which also helps with database migrations.
````
```suggestion
Note that this way of creating tables is only useful for local experimentation. For serious projects, we recommend using proper database schema migrations solution like [Alembic](https://alembic.sqlalchemy.org/en/latest/)
```

rafalp reviewed (changes requested):
Wording suggestions.

rafalp commented:
Left some notes. Sorry for taking days of delay with those, I have plenty of projects on my plate and trying to manage as much as I can until fulltime maintainer steps up for databases 🙏

empicano commented:
No worries about the delay!

I added your wording suggestions and fixed some typos 👍 

> Will the approach you are proposing also handle the constraints, keys and indexes? Or will it just create tables?

Yes. Basically, we define the SQLAlchemy schema, then compile it to a raw SQL query, then execute it with databases. The example I'm proposing to add to the documentation compiles to this (note the primary key constraint):

```
CREATE TABLE IF NOT EXISTS notes (
    id SERIAL NOT NULL, 
    text VARCHAR(100), 
    completed BOOLEAN, 
    PRIMARY KEY (id)
)
```

This compilation is a [standard part of SQLAlchemy](https://docs.sqlalchemy.org/en/20/faq/sqlexpressions.html#how-do-i-render-sql-expressions-as-strings-possibly-with-bound-parameters-inlined). When we define the database dialect (PostgreSQL, SQLite, ...) that we want to compile to -- as we do in the example -- I don't see that there are any limitations.

rafalp commented:
Looking good to me, thanks!
